climada/hazard/tc_tracks.py

```diff
@@ -210,6 +207,46 @@ def subset(self, filterdict):
 
         return out
 
+    def tracks_in_exp(self, exposure, buffer=1.0):
+        """
+        Select only the tracks that are in the vincinity (buffer) of an exposure
+        Each each exposure point/geometry is extended to a disc of
+        radius buffer. Each track is converted to a line and extended by a
+        radius buffer.
+
+        Parameters
+        ----------
+        exposure : Exposure
+            Exposure used to selectr tracks
+        buffer : float, optional
+            Size of buffer around exposure (in the units of the exposure.crs)
+            c.f. geopandas.distance.
+            The default is 1.0.
+
+        Returns
+        -------
+        filtered_tracks : climada.hazard.TCTracks()
+            TCTracks object with tracks from tc_tracks intersecting
+            the exposure whitin a buffer distance.
+
+        """
+
+        if buffer <= 0:
+            buffer = 1.0
+            LOGGER.warning(f"buffer={buffer} is invalid. buffer set to 1.0")
```

Comment:
Is there a reason for not just raising an exception if a value below zero is provided? 
```suggestion
        if buffer <= 0.0:
            raise ValueError(f"buffer={buffer} is invalid, must be above zero.")
```
What is the problem with a buffer of 0.0?

Reply:
This is based on the idea that better go to default values when obviously wrong values are given rather than ask for permission. A buffer at 0.0 is a bit of edge case where the whole geometry is collapsed to nothing instead of being actually a buffer of size 0. Hence, I do not allow for it.

Reply:
When obviously wrong values are given they should be rejected (without asking for permission). I'm opposed to auto-correction. It spoils the opportunity to fix errors early.